abdiff/core/build_ab_images.py

```diff
@@ -1 +1,67 @@
-"""abdiff.core.build_ab_images"""
+import logging
+import os
+import shutil
+
+import docker
+from pygit2 import clone_repository
+from pygit2.enums import ResetMode
+
+from abdiff.core.utils import update_or_create_job_json
+
+logger = logging.getLogger(__name__)
+
+
+def build_ab_images(
+    job_directory: str,
+    docker_client: docker.client.DockerClient,
+    commit_sha_a: str,
+    commit_sha_b: str,
+) -> tuple[str, str]:
+    """Build Docker images based on 2 commit SHAs.
+
+    Args:
+        job_directory: The directory containing all files related to a job.
+        docker_client: A configured Docker client.
+        commit_sha_a: The SHA of the first commit for comparison.
+        commit_sha_b: The SHA of the second commit for comparison.
+    """
+    images = []
+    for commit_sha in [commit_sha_a, commit_sha_b]:
+        clone_directory = f"{job_directory}/{commit_sha}"
+        image_name = f"{job_directory.split("/")[-1]}-{commit_sha}"
+        clone_repo_and_reset_to_commit(clone_directory, commit_sha)
+        image, _ = docker_client.images.build(path=clone_directory, tag=image_name)
+        images.append(image.tags[0])
+        delete_repo_clone(clone_directory)
+    images_update = {"image_name_a": images[0], "image_name_b": images[1]}
```

Comment:
Minor change request: Can we rename `images_updates` -> `images_data` to align with the naming convention for the parameter passed into `update_or_create_job_json` used for `init_job` and `init_run`? 🤔

Reply:
Good call!